decoders/path_trie.cpp

```diff
@@ -83,19 +84,28 @@ PathTrie* PathTrie::get_path_trie(int new_char, bool reset) {
   }
 }
 
-PathTrie* PathTrie::get_path_vec(std::vector<int>& output) {
-  return get_path_vec(output, ROOT_);
+PathTrie* PathTrie::get_path_vec(std::vector<int>& output, std::vector<uint32_t>* timestamps) {
+  return get_path_vec(output, ROOT_, std::numeric_limits<size_t>::max(), timestamps);
 }
 
 PathTrie* PathTrie::get_path_vec(std::vector<int>& output,
                                  int stop,
-                                 size_t max_steps) {
+                                 size_t max_steps,
+                                 std::vector<uint32_t>* timestamps) {
   if (character == stop || character == ROOT_ || output.size() == max_steps) {
     std::reverse(output.begin(), output.end());
+    if (timestamps) {
+      std::reverse(timestamps->begin(), timestamps->end());
+    }
     return this;
   } else {
     output.push_back(character);
-    return parent->get_path_vec(output, stop, max_steps);
+    if (timestamps) {
+      if (timestamps->size() == 0 || output[output.size()-1] == 0 || parent->character == ROOT_ || parent->character == 0) {
+        timestamps->push_back(offset * 20);
```

Comment:
It's better to define `20` as a parameter (for example, `ms_per_timestep`). Currently, we get 20 ms per timestep, because we use `window_stride = 10 ms` in [audio preprocessing](https://github.com/NVIDIA/OpenSeq2Seq/blob/6ebc1395f1debbc1ec93c7c6d8dcde2ab63d8781/open_seq2seq/data/speech2text/speech2text.py#L164) and stride = 2 in the first convolutional layer of Jasper. But it might be different in other ASR models.

Reply:
I have moved it out. Caller should do the conversion from offset to real time.

Reply:
Thank you, Viraj!